kernel/src/hil/storage_interface.rs

```diff
@@ -0,0 +1,74 @@
+//! Interfaces for storage devices.
+
```

Comment:
What isn't clear to me is the log interface record based, or just byte level, or something else? If I seek to position "2", does that mean the third thing written to the log? Or the third byte in the log?

Reply:
The log interface is record based, although records are identified by the position of their first byte. Seeking to position "2" will seek to the 2nd byte under the assumption that a record starts there. The log can handle seeks to invalid positions, although it won't always immediately realize that a seek was invalid.

Reply:
I see, thanks for the clarification. I read through https://github.com/tinyos/tinyos-main/blob/master/doc/txt/tep103.txt#L328, and it's not clear to me how users of the log should read from it. Are users of the log responsible for setting up some sort of delineation between log entries? The TEP doesn't go into much detail on this aspect (and even argues that the log interface is high level!), but I'm not sure if your implementation takes a different approach. I think a description of how something should use the log would be helpful.

Reply:
The log internally keeps track of the size of entries and only reads back whole entries. It will read the entries back sequentially by default, and seeking allows for reading entries out of order. However, it assumes that the user will only seek to valid entry positions (retrieved through the `current_read_offset` and `current_append_offset` functions). I could also add some additional data to the entry headers in order to better verify that a valid position was seeked if that seems necessary.

Reply:
Whatever we converge on should be documented here.